src/dymaptic.GeoBlazor.Core/Components/Layers/Graphic.cs

```diff
@@ -325,24 +337,64 @@ protected override void OnParametersSet()
         ToSerializationRecord(true);
     }
 
+    protected override async Task OnAfterRenderAsync(bool firstRender)
+    {
+        await base.OnAfterRenderAsync(firstRender);
+        GetJsModule();
+        if (JsModule is not null)
+        {
+            if (UpdateSymbol)
+            {
+                UpdateSymbol = false;
+                await SetSymbol(Symbol!);
+            }
+
+            if (_updateGeometry)
+            {
+                _updateGeometry = false;
+                await SetGeometry(Geometry!);
+            }
+            
+            if (_updatePopupTemplate)
+            {
+                _updatePopupTemplate = false;
+                await SetPopupTemplate(PopupTemplate!);
+            }
+        }
+    }
+
     private async void OnAttributesChanged(object? sender, NotifyCollectionChangedEventArgs e)
     {
-        if (JsGraphicReference is null) return;
+        GetJsModule();
+        if (JsModule is null) return;
 
-        await JsGraphicReference.InvokeVoidAsync("setAttributes", 
-            CancellationTokenSource.Token, Attributes);
+        await JsModule.InvokeVoidAsync("setGraphicAttributes", 
+            CancellationTokenSource.Token, Id, Attributes);
         ToSerializationRecord(true);
     }
+
+    private void GetJsModule()
```

Comment:
could you override the JSModule property and call this in the get body of it? then you don't need to explicitely call it. This also may not work with all of the serialization stuff going on

Reply:
I thought of that, I'll take another look. See my other answer for why this is required in `Graphic` but not elsewhere.